WordPress/src/main/java/org/wordpress/android/ui/sitecreation/domains/SiteCreationDomainValidator.kt

```diff
@@ -0,0 +1,56 @@
+package org.wordpress.android.ui.sitecreation.domains
+
+import org.wordpress.android.util.UrlUtilsWrapper
+import java.net.URL
+import javax.inject.Inject
+
+class SiteCreationDomainValidator
+@Inject constructor(
+    private val urlUtilsWrapper: UrlUtilsWrapper,
+    private val suffixDatabase: PublicSuffixDatabaseWrapper
+) {
+    /**
+     * Checks to see if the supplied query is a valid domain. If not then it could be a word, so a WP.com
+     * sub domain is created with it and then that's checked. If none of those are valid then we assume the supplied
+     * query isn't a valid domain for site creation.
+     */
+    fun validateDomain(query: String): ValidationResult {
+        var isValidUrl: Boolean
+
+        var fullUrl = urlUtilsWrapper.addUrlSchemeIfNeeded(query, false)
+
+        isValidUrl = urlUtilsWrapper.isValidUrlAndHostNotNull(fullUrl) && suffixDatabase.getEffectiveTldPlusOne(
```

Comment:
@malinajirka what I attempted to do in this class is to reuse the `URLUtils` that existed in the project to perform the domain validation. However, on it's own it wasn't verifying that a domain would contain a TLD so I ended up using the PublicSuffixDatabase that's shipped with `OkHttp` not sure how you feel about this, but it was the easier solution so I definitely want some feedback on this.

Reply:
I am just commenting here to finalize these discussions we had started and somewhat concluded in the past. The `PublicSuffixDatabase` is an actively updated database with TLDs that being used in the world right now. Nonetheless, it's an overkill for this solution so it will be removed.